### PR TITLE
Go to Definition for razor projection buffers

### PR DESCRIPTION
**Pull request by KirillOsenkov**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1010683

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Documents/DocumentManager.cs [hunk 1]
```diff
@@ -253,16 +253,17 @@ public async Task<Document> OpenDocument (DocumentController controller, bool br
 
 		public async Task<Document> OpenDocument (FileOpenInformation info)
 		{
-			if (string.IsNullOrEmpty (info.FileName))
+			FilePath filePath = GetActualFilePathToOpen(info.FileName);
```
Comment (sandyarmstrong) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Documents/DocumentManager.cs]: Does this prevent the user from ever opening the webtools `*.g.cs` files? Do they expect to have a way to do that?

**Comment by sandyarmstrong** [comment]
@monojenkins backport release-8.4

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Documents/DocumentManager.cs (hunk 1 above)
Comment (KirillOsenkov) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Documents/DocumentManager.cs]: You're right, I don't like this. I only want to do this on the codepath where the workspace is asking to open the .g.cs file.

I'll redo it.

**Comment by KirillOsenkov** [comment]
OK we think this is in a good shape now

**Comment by KirillOsenkov** [comment]
@monojenkins backport release-8.4

**Pull request merged**